### PR TITLE
chore: Ensure that CHE_INTEGRATION_XXXX_SERVER__ENDPOINTS and CHE_INT…

### PR DESCRIPTION
**Pull request by tolusha**
…EGRATION_XXXX_OAUTH__ENDPOINT properties are properly set

Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### What does this PR do?
Ensure that CHE_INTEGRATION_XXXX_SERVER__ENDPOINTS and CHE_INTEGRATION_XXXX_OAUTH__ENDPOINT properties are properly set. 
Changes:
* CHE_OAUTH_GITLAB_CLIENTID__FILEPATH --> CHE_OAUTH2_GITLAB_CLIENTID__FILEPATH
* CHE_OAUTH_GITLAB_CLIENTSECRET__FILEPATH --> CHE_OAUTH2_GITLAB_CLIENTSECRET__FILEPATH

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20902

### How to test this PR?

 
1. Deploy and create secrets
```bash
$ chectl server:deploy \
     --installer operator \
     --platform minikube \
     --che-operator-image abazko/operator:20902
     
$ kubectl apply -f - <<EOF
apiVersion: v1
data:
  private.key: cHJpdmF0ZSBrZXkgZGF0YQ==
  consumer.key: Y29uc3VtZXIga2V5IGRhdGE=
kind: Secret
metadata:
  annotations:
    che.eclipse.org/oauth-scm-server: bitbucket
    che.eclipse.org/scm-server-endpoint: bitbucket_endpoint_1
  labels:
    app.kubernetes.io/component: oauth-scm-configuration
    app.kubernetes.io/part-of: che.eclipse.org
  name: bitbucket-oauth-config
  namespace: eclipse-che
EOF

$ kubectl apply -f - <<EOF
apiVersion: v1
data:
  id: aWQ=
  secret: c2VjcmV0
kind: Secret
metadata:
  annotations:
    che.eclipse.org/oauth-scm-server: github
    che.eclipse.org/scm-server-endpoint: github_endpoint_1
  labels:
    app.kubernetes.io/component: oauth-scm-configuration
    app.kubernetes.io/part-of: che.eclipse.org
  name: github-oauth-config
  namespace: eclipse-che
EOF

$ kubectl apply -f - <<EOF
apiVersion: v1
data:
  id: aWQgZGF0YQ==
  secret: c2VjcmV0IGRhdGE=
kind: Secret
metadata:
  annotations:
    che.eclipse.org/oauth-scm-server: gitlab
    che.eclipse.org/scm-server-endpoint: gitlab_endpoint_1
  labels:
    app.kubernetes.io/component: oauth-scm-configuration
    app.kubernetes.io/part-of: che.eclipse.org
  name: gitlab-oauth-config
  namespace: eclipse-che
EOF
```

2. Wait some time and check env
```
$ SERVER_POD=$(kubectl get pod -l app.kubernetes.io/component=che -n eclipse-che --no-headers=true | awk '{print $1}')
$ kubectl exec -it $SERVER_POD -n eclipse-che  -- bash  -c "env | grep _ENDPOINT | sort"
CHE_INTEGRATION_BITBUCKET_SERVER__ENDPOINTS=bitbucket_endpoint_1
CHE_INTEGRATION_GITHUB_OAUTH__ENDPOINT=github_endpoint_1
CHE_INTEGRATION_GITHUB_SERVER__ENDPOINTS=github_endpoint_1
CHE_INTEGRATION_GITLAB_OAUTH__ENDPOINT=gitlab_endpoint_1
CHE_INTEGRATION_GITLAB_SERVER__ENDPOINTS=gitlab_endpoint_1
CHE_OAUTH1_BITBUCKET_ENDPOINT=bitbucket_endpoint_1
```

3. Update CheCluster CR
```
$ kubectl patch checluster/eclipse-che --patch "{\"spec\":{\"server\":{\"customCheProperties\": {\"CHE_INTEGRATION_GITHUB_SERVER__ENDPOINTS\": \"github_endpoint_2,github_endpoint_3\"}}}}" --type=merge -n eclipse-che
$ kubectl patch checluster/eclipse-che --patch "{\"spec\":{\"server\":{\"customCheProperties\": {\"CHE_INTEGRATION_GITLAB_SERVER__ENDPOINTS\": \"gitlab_endpoint_2,gitlab_endpoint_3\"}}}}" --type=merge -n eclipse-che
$ kubectl patch checluster/eclipse-che --patch "{\"spec\":{\"server\":{\"customCheProperties\": {\"CHE_INTEGRATION_BITBUCKET_SERVER__ENDPOINTS\": \"bitbucket_endpoint_2,bitbucket_endpoint_3\"}}}}" --type=merge -n eclipse-che
```

4. Wait some time and check env
```
$ SERVER_POD=$(kubectl get pod -l app.kubernetes.io/component=che -n eclipse-che --no-headers=true | awk '{print $1}')
$ kubectl exec -it $SERVER_POD -n eclipse-che  -- bash  -c "env | grep _ENDPOINT | sort"
CHE_INTEGRATION_BITBUCKET_SERVER__ENDPOINTS=bitbucket_endpoint_1,bitbucket_endpoint_2,bitbucket_endpoint_3
CHE_INTEGRATION_GITHUB_OAUTH__ENDPOINT=github_endpoint_1
CHE_INTEGRATION_GITHUB_SERVER__ENDPOINTS=github_endpoint_1,github_endpoint_2,github_endpoint_3
CHE_INTEGRATION_GITLAB_OAUTH__ENDPOINT=gitlab_endpoint_1
CHE_INTEGRATION_GITLAB_SERVER__ENDPOINTS=gitlab_endpoint_1,gitlab_endpoint_2,gitlab_endpoint_3
CHE_OAUTH1_BITBUCKET_ENDPOINT=bitbucket_endpoint_1
```

5. Check mounts
```
$ kubectl exec -it $SERVER_POD -n eclipse-che  -- bash  -c "env | grep PATH | sort"
CHE_OAUTH1_BITBUCKET_CONSUMERKEYPATH=/che-conf/oauth/bitbucket/consumer.key
CHE_OAUTH1_BITBUCKET_PRIVATEKEYPATH=/che-conf/oauth/bitbucket/private.key
CHE_OAUTH2_GITHUB_CLIENTID__FILEPATH=/che-conf/oauth/github/id
CHE_OAUTH2_GITHUB_CLIENTSECRET__FILEPATH=/che-conf/oauth/github/secret
CHE_OAUTH2_GITLAB_CLIENTID__FILEPATH=/che-conf/oauth/gitlab/id
CHE_OAUTH2_GITLAB_CLIENTSECRET__FILEPATH=/che-conf/oauth/gitlab/secret

$ kubectl exec -it $SERVER_POD -n eclipse-che  -- bash  -c "ls -R /che-conf/oauth"
/che-conf/oauth:
bitbucket  github  gitlab

/che-conf/oauth/bitbucket:
consumer.key  private.key

/che-conf/oauth/github:
id  secret

/che-conf/oauth/gitlab:
id  secret
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by mshaposhnik — APPROVED** [review]

File: pkg/deploy/server/server_configmap.go [hunk 1]
```diff
@@ -326,34 +326,40 @@ func (s *CheServerReconciler) getCheConfigMapData(ctx *deploy.DeployContext) (ch
...
+	if err != nil {
+		return nil, err
+	}
+
+	err = updateIntegrationServerEndpoints(ctx, cheEnv, "gitlab")
+	if err != nil {
+		return nil, err
+	}
+
+	err = updateIntegrationServerEndpoints(ctx, cheEnv, "github")
```
Comment (mmorhun) [inline pkg/deploy/server/server_configmap.go:339]: loop?

File: pkg/deploy/server/server_deployment.go [hunk 2]
```diff
@@ -404,115 +403,58 @@ func GetFullCheServerImageLink(checluster *orgv1.CheCluster) string {
...
-		}
 	}
 
+	mountVolumes(deployment, secret, deploy.BitBucketOAuthConfigMountPath)
+	mountEnv(deployment, "CHE_OAUTH1_BITBUCKET_CONSUMERKEYPATH", deploy.BitBucketOAuthConfigMountPath+"/"+deploy.BitBucketOAuthConfigConsumerKeyFileName)
+	mountEnv(deployment, "CHE_OAUTH1_BITBUCKET_PRIVATEKEYPATH", deploy.BitBucketOAuthConfigMountPath+"/"+deploy.BitBucketOAuthConfigPrivateKeyFileName)
+
+	oauthEndpoint := secret.Annotations[deploy.CheEclipseOrgScmServerEndpoint]
+	if oauthEndpoint != "" {
+		mountEnv(deployment, "CHE_OAUTH1_BITBUCKET_ENDPOINT", oauthEndpoint)
```
Comment (mmorhun) [inline pkg/deploy/server/server_deployment.go:418]: Is it supposed to have only one `_` before `ENDPOINT`?

**Review by mmorhun — APPROVED** [review]

File: pkg/deploy/server/server_deployment.go (hunk 2 above)
Comment (tolusha) [reply pkg/deploy/server/server_deployment.go:418]: Right, only one `_`

File: pkg/deploy/server/server_configmap.go (hunk 1 above)
Comment (tolusha) [reply pkg/deploy/server/server_configmap.go]: agree

**Pull request merged**